qa/os/src/test/java/org/elasticsearch/packaging/test/EnrollNodeToClusterTests.java

```diff
@@ -0,0 +1,332 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.packaging.test;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.ssl.PemUtils;
+import org.elasticsearch.core.PathUtils;
+import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.packaging.util.Archives;
+import org.elasticsearch.packaging.util.Installation;
+import org.elasticsearch.packaging.util.ServerUtils;
+import org.elasticsearch.packaging.util.Shell;
+import org.elasticsearch.test.http.MockRequest;
+import org.elasticsearch.test.http.MockResponse;
+import org.elasticsearch.test.http.MockWebServer;
+import org.elasticsearch.xpack.core.security.EnrollmentToken;
+import org.elasticsearch.xpack.core.ssl.TestsSSLService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.packaging.util.Archives.ARCHIVE_OWNER;
+import static org.elasticsearch.packaging.util.Archives.installArchive;
+import static org.elasticsearch.packaging.util.Archives.verifyArchiveInstallation;
+import static org.elasticsearch.packaging.util.FileMatcher.Fileness.Directory;
+import static org.elasticsearch.packaging.util.FileMatcher.Fileness.File;
+import static org.elasticsearch.packaging.util.FileMatcher.file;
+import static org.elasticsearch.packaging.util.FileMatcher.p660;
+import static org.elasticsearch.packaging.util.FileMatcher.p750;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeTrue;
+
+public class EnrollNodeToClusterTests extends PackagingTestCase {
+
+    private static MockWebServer esNode;
+
+    @BeforeClass
+    public static void filterDistros() {
+        assumeTrue("only archives", distribution.isArchive());
+    }
+
+    /**
+     * We mock an enrollment response from an elasticsearch node. What we want to test in this class is that the startup scripts
+     * can correctly parse the parameters from the command line and that the enrollment can be invoked as expected. The enrollment
+     * process is tested in other tests so we don't care about duplicating this here and attempt to spin up multiple nodes and
+     * do a proper enrollment.
+     * We do need to have TLS because the enrolling process is hardcoded to work only with HTTP over TLS and to trust certificates
+     * that are signed by the certificate identified with the fingerprint in the enrollment token.
+     */
+    @Before
+    public void setupMockEsNode() throws Exception {
+        String enrollmentResponseBody = "{\"nodes_addresses\":[\"127.0.0.1:9300\", \"192.168.1.10:9301\"], "
```

Comment:
nit: Maybe this can be reconstituted from the resource files, or maybe it can become itself a resource.